File: AspNetMvcUnleashed/0672329980_source/Chapter14/CS/MvcApplication1/MvcApplication1/Controllers/MovieProgressController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using System.Web.Mvc.Ajax;
using MvcApplication1.Models;
using System.Threading;

namespace MvcApplication1.Controllers
{
    public class MovieProgressController : Controller
    {

        private MoviesDBEntities _entities = new MoviesDBEntities();

        // GET: /Movie/

        public ActionResult Index()
        {
            return View(_entities.MovieSet.ToList());
        }

        // GET: /Movie/Create

        public ActionResult Create()
        {
            return View();
        }

        // POST: /Movie/Create

        [AcceptVerbs(HttpVerbs.Post)]
        public string Create(Movie movieToCreate)
        {
            Thread.Sleep(5 * 1000);
            try
            {
                _entities.AddToMovieSet(movieToCreate);
                _entities.SaveChanges();
                return "Inserted new movie " + movieToCreate.Title;
            }
            catch
            {
                return "Could not insert movie " + movieToCreate.Title;
            }
        }


    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter14/VB/MvcApplication1/MvcApplication1/Controllers/MovieProgressController.vb
Imports System.Threading

Public Class MovieProgressController
    Inherits Controller

    Private _entities As New MoviesDBEntities()

    ' GET: /Movie/

    Public Function Index() As ActionResult
        Return View(_entities.MovieSet.ToList())
    End Function

    ' GET: /Movie/Create

    Public Function Create() As ActionResult
        Return View()
    End Function

    ' POST: /Movie/Create

    <AcceptVerbs(HttpVerbs.Post)> _
    Public Function Create(ByVal movieToCreate As Movie) As String
        Thread.Sleep(5 * 1000)

        Try
            _entities.AddToMovieSet(movieToCreate)
            _entities.SaveChanges()
            Return "Inserted new movie " & movieToCreate.Title
        Catch
            Return "Could not insert movie " & movieToCreate.Title
        End Try
    End Function

End Class
